recipe-generator-node/src/index.js

```diff
@@ -0,0 +1,118 @@
+const { ArgumentParser } = require("argparse");
+const { exec } = require("child_process");
+const http = require("http");
+const fs = require("fs");
+
+const RECIPE_VERSION = "0.3.4";
+const GENERATOR_URL = `http://search.maven.org/remotecontent?filepath=ca/derekcormier/recipe/recipe-generator/0.3.4/recipe-generator-${RECIPE_VERSION}-jar-with-dependencies.jar`;
+
+main();
+
+async function main() {
+    const options = parseArgs();
+    try {
+        await downloadGeneratorJar(options);
+    }
+    catch(err) {
+        console.log(err);
+    }
+    runGeneration(options);
+}
+
+function parseArgs() {
+    const parser = new ArgumentParser({
+        addHelp: true
+    });
+    parser.addArgument("domain", {
+        help: "name of ingredient domain",
+        type: "string"
+    });
+    parser.addArgument("flavour", {
+        help: "type of generation to perform; valid options: java-ingredient, java-hook, js-ingredient, ts-ingredient",
+        type: "string"
+    });
+    parser.addArgument("cookbook", {
+        help: "path to the yaml cookbook definition file",
+        type: "string"
+    });
+    parser.addArgument("targetDir", {
+        help: "directory to output generated files",
+        type: "string"
+    });
+    parser.addArgument("--javaPackage", {
+        help: "java package for generated classes",
+        type: "string",
+        metavar: "package"
+    });
+    parser.addArgument("--ingredientPostfix", {
+        help: "string to append to ingredient names",
+        type: "string",
+        metavar: "postfix"
+    });
+    parser.addArgument("--jarDownloadDir", {
+        help: "directory to put the downloaded recipe generator jar",
+        type: "string",
+        defaultValue: "generator",
+        metavar: "directory"
+    });
+    
+    return parser.parseArgs();
+}
+
+async function downloadGeneratorJar(options) {
+    if (!fs.existsSync(options.jarDownloadDir)) {
```

Comment:
Do you need to use the `*Sync` version of methods when the method this is in is `async`? 🤔